src/interactive-window/debugger/jupyter/kernelDebugAdapter.ts

```diff
@@ -67,15 +68,19 @@ export class KernelDebugAdapter extends KernelDebugAdapterBase {
         }
     }
 
-    override handleMessage(message: DebugProtocol.ProtocolMessage): Promise<KernelMessage.IDebugReplyMsg | undefined> {
+    override handleMessage(message: DebugProtocol.ProtocolMessage) {
         traceInfoIfCI(`KernelDebugAdapter::handleMessage ${JSON.stringify(message, undefined, ' ')}`);
         if (message.type === 'request' && this.debugLocationTracker?.onWillReceiveMessage) {
             this.debugLocationTracker.onWillReceiveMessage(message);
         }
         if (message.type === 'response' && this.debugLocationTracker?.onDidSendMessage) {
             this.debugLocationTracker.onDidSendMessage(message);
         }
-        return super.handleMessage(message);
+        const promise = super.handleMessage(message);
+        // The VS Code debugger class does not support an async `handleMessage`, its supposed to be sync.
+        // As a result, any errors here will not be handled and they'll be treated as unhandled errors.
+        promise.catch(noop);
```

Comment:
Thanks, but it looks like there is already a try/catch around handleMessage in the base class, right? Even so, would you want to log the error?

Reply:
oh, didn't realise that, will close this then.
i guess i added it at some point then